tests/src/Unit/Plugin/Commerce/PriceRuleCalculation/PriceListTest.php

```diff
@@ -0,0 +1,270 @@
+<?php
+
+namespace Drupal\Tests\commerce_price_rule\Unit\Plugin\Commerce\PriceRuleCalculation;
+
+use Drupal\commerce\Context;
+use Drupal\commerce_price\Entity\CurrencyInterface;
+use Drupal\commerce_price\Price;
+use Drupal\commerce_price\Rounder;
+use Drupal\commerce_price_rule\Entity\PriceListInterface;
+use Drupal\commerce_price_rule\Entity\PriceRuleInterface;
+use Drupal\commerce_price_rule\Plugin\Commerce\PriceRuleCalculation\PriceList;
+use Drupal\commerce_product\Entity\ProductVariationInterface;
+use Drupal\commerce_store\Entity\StoreInterface;
+
+use Drupal\Core\Database\Connection;
+use Drupal\Core\Database\Driver\sqlite\Statement;
+use Drupal\Core\Database\Query\SelectInterface;
+use Drupal\Core\Entity\EntityManagerInterface;
+use Drupal\Core\Entity\EntityStorageInterface;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Tests\UnitTestCase;
+use Drupal\user\UserInterface;
+
+/**
+ * Class PriceListTest.
+ *
+ * Tests the price list price rule calculation for commerce price rules.
+ *
+ * @coversDefaultClass \Drupal\commerce_price_rule\Plugin\Commerce\PriceRuleCalculation\PriceList
+ * @group commerce_price_rule
+ * @package Drupal\Tests\commerce_price_rule\Unit
+ */
+class PriceListTest extends UnitTestCase {
+
+  /**
+   * Provides data for testCalculate().
+   *
+   * @return array
+   *   The providers.
+   */
+  public function calculateProviderConfiguration() {
+    // Create a mock database connection.
+    $statement = $this->prophesize(Statement::class);
+    $statement->fetchAssoc()->willReturn($this->fetchAssocResults());
+    $statement = $statement->reveal();
+
+    $query = $this->prophesize(SelectInterface::class);
+    $query->fields('li', ['price__number', 'price__currency_code'])->willReturn($this->returnSelf());
+    /*$query->condition('li.price_list_id', 1)
```

Comment:
@krystalcode This is still an issue. The code works fine if I comment out most of the query conditions in the actual PriceList module:

```php
    // Performance is important here, load only the required fields directly
    // from the database.
    $query = $this->database_connection
      ->select('commerce_price_rule_list_item', 'li');
      /*->fields('li', ['price__number', 'price__currency_code'])
      ->condition('li.price_list_id', $this->configuration['price_list_id'])
      ->condition('li.product_variation_id', $entity->id())
      ->condition('li.status', 1);
      ->range(0, 1);

    // The quantity must be matching the minimum and maximum quantites of the
    // list item i.e. it should either be both higher or equal than the minimum
    // and lower or equal than the maximum. When no maximum is defined, which
    // should happen at the last list item for a product varation (e.g. price X
    // for more than Y number of items), the quantity only needs to be higher
    // than the minimum.
    $quantity_max_defined = $query->andConditionGroup()
      ->condition('min_quantity', $quantity, '<=')
      ->condition('max_quantity', $quantity, '>=');
    $quantity_max_undefined = $query->andConditionGroup()
      ->condition('min_quantity', $quantity, '<=')
      ->condition('max_quantity', NULL, 'IS NULL');
    $quantity_query = $query->orConditionGroup()
      ->condition($quantity_max_defined)
      ->condition($quantity_max_undefined);
    $query->condition($quantity_query);*/

    $result = $query
      ->execute()
      ->fetchAssoc();
```

I'm just not sure what the issue is here.